integration-tests/tests/src/tests/sync/user.ts

```diff
@@ -39,17 +39,17 @@ function expectIsUSer(user: Realm.User) {
   expect(user).instanceOf(Realm.User);
 }
 
-function expectIsSameUser(value: Realm.User, user: Realm.User | null) {
+function expectIsSameUser(value: Realm.User, user: Realm.User<any, any> | null) {
```

Comment:
This and the following ones are just my "I don't know what I'm doing" attempt at making typescript stop complaining about the types. I'm curious to know what would be the correct approach for similar cases

Reply:
Hmm, when you hover over a user, the type isn't `Realm.User`. I think that's the root cause.

<img width="1016" alt="Screenshot 2023-06-26 at 12 51 38" src="https://github.com/realm/realm-js/assets/1225363/e8985415-5336-4b31-82ef-a8eba1f69c95">

Reply:
We do have an `AnyUser` you could use as well.

Reply:
@kneth, yes, because it's using generics. 
@takameyer I can see that, but it seems that's not exported by Realm, so I'd keep it like this if it makes sense?